addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js

```diff
@@ -262,6 +265,46 @@ export class GlobalFiltersUIPlugin extends spreadsheet.UIPlugin {
         }
     }
 
+    /**
+     * Returns the possible values a text global filter can take
+     * if the values are restricted by a range of allowed values
+     * @param {string} filterId
+     * @returns {{value: any, formattedValue: string}[]}
+     */
+    getTextGlobalFilterAllowedValues(filterId) {
+        const filter = this.getters.getGlobalFilter(filterId);
+        const range = filter.rangeOfAllowedValues;
+        if (!range) {
+            return [];
+        }
+        const cells = this.getters.getEvaluatedCellsInZone(range.sheetId, range.zone);
+
+        const uniqueValues = new Set();
+        const allowedValues = [];
+        for (const cell of cells) {
+            const formattedValue = cell.formattedValue;
+            if (!["empty", "error"].includes(cell.type) && !uniqueValues.has(formattedValue)) {
+                uniqueValues.add(formattedValue);
+                allowedValues.push({
+                    value: cell.value,
+                    formattedValue,
+                });
+            }
+        }
+        if (filter.defaultValue && !uniqueValues.has(filter.defaultValue)) {
+            uniqueValues.add(filter.defaultValue);
+            allowedValues.push({ value: filter.defaultValue, formattedValue: filter.defaultValue });
+        }
+        // the current value might not be in the range
+        // if the range cells changed in the meantime
+        const currentValue = this.getGlobalFilterValue(filterId);
+        if (currentValue && !uniqueValues.has(currentValue)) {
+            uniqueValues.add(currentValue);
+            allowedValues.push({ value: currentValue, formattedValue: currentValue });
+        }
```

Comment:
probably possible do everything in a single loop, without 2 extra if for default and current values, if you care about writing short code.

Someing like array = cell.filter(notempty).map(toValueAndFormattedValue); array.concat(currentValue, defaultValue); ...